eZ/Publish/Core/SignalSlot/RoleService.php

```diff
@@ -66,8 +67,243 @@ public function __construct(RoleServiceInterface $service, SignalDispatcher $sig
+    {
+        $returnValue = $this->service->createRoleDraft($roleCreateStruct);
+        $this->signalDispatcher->emit(
+            //TODO: CreateRoleDraftSignal? Or skip it altogether?
+            new CreateRoleSignal(
+                array(
+                    'roleId' => $returnValue->id,
+                )
+            )
+        );
+
+        return $returnValue;
+    }
+
+    /**
+     * Creates a new RoleDraft for existing Role.
+     *
+     * @since 6.0
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to create a role
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException if the Role already has a Role Draft that will need to be removed first
+     * @throws \eZ\Publish\API\Repository\Exceptions\LimitationValidationException if a policy limitation in the $roleCreateStruct is not valid
+     *
+     * @param \eZ\Publish\API\Repository\Values\User\Role $role
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\RoleDraft
+     */
+    public function createRoleDraftByRole(Role $role)
+    {
+        $returnValue = $this->service->createRoleDraftByRole($role);
+        $this->signalDispatcher->emit(
+            //TODO: CreateRoleDraftSignal? Or skip it altogether?
+            new CreateRoleSignal(
+                array(
+                    'roleId' => $returnValue->id,
+                )
+            )
+        );
+
+        return $returnValue;
+    }
+
+    /**
+     * Loads a role for the given id.
+     *
+     * @since 6.0
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to read this role
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException if a role with the given id was not found
+     *
+     * @param mixed $id
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\RoleDraft
+     */
+    public function loadRoleDraft($id)
+    {
+        return $this->service->loadRoleDraft($id);
+    }
+
+    /**
+     * Updates the properties of a role draft.
+     *
+     * @since 6.0
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to update a role
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException if the identifier of the role already exists
+     *
+     * @param \eZ\Publish\API\Repository\Values\User\RoleDraft $roleDraft
+     * @param \eZ\Publish\API\Repository\Values\User\RoleUpdateStruct $roleUpdateStruct
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\RoleDraft
+     */
+    public function updateRoleDraft(RoleDraft $roleDraft, RoleUpdateStruct $roleUpdateStruct)
+    {
+        $returnValue = $this->service->updateRoleDraft($roleDraft, $roleUpdateStruct);
+        $this->signalDispatcher->emit(
+            //TODO: UpdateRoleDraftSignal? Or skip it altogether?
```

Comment:
Can you actually update a Role that ain't a draft ? It would not make much sense, would it ?